apps/opik-backend/src/main/java/com/comet/opik/api/sorting/SortingFactory.java

```diff
@@ -0,0 +1,62 @@
+package com.comet.opik.api.sorting;
+
+import com.comet.opik.utils.JsonUtils;
+import com.fasterxml.jackson.core.type.TypeReference;
+import jakarta.ws.rs.BadRequestException;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public abstract class SortingFactory {
+    public static final String ERR_INVALID_SORTING_PARAM_TEMPLATE = "Invalid sorting query parameter '%s'";
+    public static final String ERR_ILLEGAL_SORTING_FIELDS_TEMPLATE = "Invalid sorting fields '%s'";
+    public List<SortingField> newSorting(String queryParam) {
+        if (StringUtils.isBlank(queryParam)) {
+            return null;
+        }
+
+        List<SortingField> sorting;
+        try {
+            sorting = JsonUtils.readValue(queryParam, new TypeReference<>() {
+            });
+        } catch (UncheckedIOException exception) {
+            throw new BadRequestException(ERR_INVALID_SORTING_PARAM_TEMPLATE.formatted(queryParam), exception);
+        }
+
+        if (sorting.isEmpty()) {
+            return null;
+        }
+
+        validateFields(sorting);
+
+        return sorting;
+    }
+
+    public String toOrderBySql(List<SortingField> sorting) {
```

Comment:
I would not keep this method here as this factory should only validate and parse the API fields